### ISSUE TITLE
Add support to deploy other objects with Custom Serving Runtime Templates

### ISSUE THREAD

**Issue by lucferbux** [opendatahub-io/odh-dashboard#1190]
### Feature description

In Custom Serving Runtimes v1 we allow adding custom yaml specifications for Serving Ruttimes (see [mocks](https://www.sketch.com/s/5c26573b-fd37-4d54-a4f7-aab1cc17ff5d/a/P1GEp08)).
We want to allow users to deploy additional resources (ConfiMaps...) when selecting a Custom Serving Runtime.

**Comment by lucferbux** [comment]
@vconzola Let me know your thoughts on this, one implementation could be use the yaml editor to add multiple files, we must define the scope of files we wanna deploy for that but l want feedback from UX.

**Comment by vconzola** [comment]
@lucferbux Is this requirement documented someplace? And is this something that we'd want only for resources related to serving runtimes or might it be something we want to be more general? That will help determine where and how this feature fits into the UI. Finally, is there a target date for this feature?

**Comment by lucferbux** [comment]
Hi @vconzola no, this is not documented, was tracked from a conversation with @cfchase he told us that one of the requests that they were asked. I don't have further information, so I'm not sure which files are needed. And we don't have a target date defined, in that conversation we talked about next sprint.

**Comment by lucferbux** [comment]
@cfchase If you can give us a little bit more context it would be helpful to triage the task. Thanks in advance.

**Comment by cfchase** [comment]
Yes, some serving runtimes (model server) require creating an associated configmap which is referenced from the generated resource.  So as part of the template, you'd need both objects.

**Comment by vconzola** [comment]
@cfchase How do we know if a serving runtime requires a config map? Is there something in the runtime YAML that indicates this? Also, is the config map definition in the same YAML file as the runtime, or in a separate file?

**Comment by lucferbux** [comment]
@vconzola YAML specification allows to define several objects in a file with the `---` separator, you can see examples [here](https://github.com/opendatahub-io/odh-manifests/blob/master/model-mesh/dependencies/quickstart.yaml#L28). I guess this is the most straightforward way to support it, but I'm not sure if we need to adapt the UX to tell the user that that's supported. Other option is to treat those files as the env variables in workbenches (which we currently support for configmaps and secrets).

Also, my opinion here is that we should limit the kind of objects we want to support, @cfchase is there any other resource rather than a configmap needed, should we allow store more than two resources (serving runtime + configmap) in the template? I would want to limit the scope of this since the logic behind can increase in complexity really quickly.
